### PR TITLE
Add server write and client<-> server interaction tests

### PR DESCRIPTION
**Pull request by gaborigloi**
* Test that the server handles writes correctly
* Add a new kind of test that connects together the server and client using a pipe and checks their interaction

Signed-off-by: Gabor Igloi <gabor.igloi@citrix.com>

### CODE REVIEW

**Comment by mseri** [comment]
Careful with lwt_log. It is delrecated and now requires to add the lwt_log package in the opam file

**Comment by gaborigloi** [comment]
Yes, but we already use it in this library - I didn't want to move to `logs` in this PR, as that's a lot of work

**Comment by mseri** [comment]
It's fine, but we need to update the opam file (and possibly some jbuild file). They will break at the next lwt update

**Comment by gaborigloi** [comment]
We don't yet have the `lwt_log` package in xs-opam, so we'll have to add it there first.
So shall we get this in, and when the package is also in, we can update all the affected libraries?

**Comment by mseri** [comment]
Fine with me. I'll review it asap

**Comment by gaborigloi** [comment]
Thanks! I've opened https://github.com/xapi-project/xs-opam/pull/235 for adding `lwt_log`

File: lib_test/client_server_test.ml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    process. *)
+
+open Lwt.Infix
+
+let with_channels f =
+  let section = Lwt_log_core.Section.make("with_channels") in
+  let make_channel name (ic, oc) =
+    let read c =
+      Lwt_log.debug_f ~section "%s read: %d" name (Cstruct.len c) >>= fun () ->
+      let len = Cstruct.len c in
```
Comment (mseri) [inline lib_test/client_server_test.ml]: If you move it the line above you can avoid duplivating the call. Similarly in write c

**Comment by lindig** [comment]
Can we make the one requested change and then merge it?

**Comment by gaborigloi** [comment]
@lindig Yep, I compeltely forgot about this

**Pull request merged**